Fix Sonar-reported TypeScript code smells and minor bugs

jonasiu opened the pull request:
* Added a Code Churn Analysis section and API details to the README.
* Update Node core imports to `node:*` variants across backend and frontend.
* Replace `parseInt`/`isNaN` with `Number.parseInt`/`Number.isNaN` and use `??` where appropriate.
* Prefer `for…of`, `Set`, `globalThis`, and clearer error types; remove redundant `undefined` and zero fractions.
* Refactor high-complexity functions and fix an invalid loop in `fileAnalysisService` plus other Sonar rule violations in `gitService`, `repositoryCache`, `metrics`, `commitRoutes`, and related modules.
* All listed Sonar issues are now marked as fixed.

jonasyr committed:
Updated Claude.md

jonasyr committed:
feat(docs): add Code Churn Analysis section and API details to README

jonasyr committed:
Refactor code structure for improved readability and maintainability

jonasyr committed:
Refactor code structure for improved readability and maintainability

jonasyr committed:
Fixed some Minor Sonarqube Issues

jonasyr committed:
Fixed Sonarqube Issues

jonasyr committed:
Revert "Fixed Sonarqube Issues"

This reverts commit 829e36b064dcf4effdd14faeb9d80d7968b779c7.

jonasyr committed:
Fixed SOnarqube Issues

jonasyr committed:
Updated Browser Mapping

jonasyr committed:
Fixed SOnarqube Issues

jonasyr committed:
Fixed SOnarqube Issues

jonasyr committed:
Removed Issue csv

jonasyr commented on apps/frontend/src/pages/MainPage.tsx:
```diff
@@ -53,6 +53,9 @@ const MainPage: React.FC = () => {
     }
   };
 
+  const canShowVisualizations =
+    isLoading === false && error == null && commits.length > 0;
+
```
Will be replaced soon

jonasyr committed:
- Added a reusable logValidationMessages helper so configuration validation now logs errors and warnings consistently before throwing, keeping the console output structured and reducing nesting in handleValidationResults.
- Simplified GitService.buildCommitStatsArgs to bail out early when filters are absent, normalize author filters, and tightened the churn-threshold unit test to remove the unused dataset that triggered lint warnings.
- Added a focused cache-endpoint test suite that exercises validation failures, the /cache/stats response, and cache invalidation behavior with mocked dependencies, ensuring the new route logic is covered without invoking real git operations.
